src/handler.rs

```diff
@@ -105,13 +118,28 @@ impl axum::response::IntoResponse for Error {
     }
 }
 
+type Result<T> = std::result::Result<T, Error>;
+
+fn get_server_from_state(
+    state: &OPRFState,
+    instance_name: String,
+) -> Result<RwLockReadGuard<'_, OPRFInstance>> {
+    Ok(state
+        .instances
+        .get(&instance_name)
+        .ok_or(Error::InstanceNotFound(instance_name))?
+        .read()?)
+}
+
 /// Process PPOPRF evaluation requests
-pub async fn randomness(
-    State(state): State<OPRFState>,
-    Json(request): Json<RandomnessRequest>,
-) -> Result<Json<RandomnessResponse>, Error> {
+#[instrument(skip(state, request))]
+async fn randomness(
+    state: OPRFState,
+    instance_name: String,
+    request: RandomnessRequest,
+) -> Result<Json<RandomnessResponse>> {
     debug!("recv: {request:?}");
-    let state = state.read()?;
+    let state = get_server_from_state(&state, instance_name)?;
     let epoch = request.epoch.unwrap_or(state.epoch);
     if epoch != state.epoch {
```

Comment:
The requested epoch needs to checked against instance.epoch.

Reply:
`state` is the instance state, which we check against on this line

Reply:
Ah you're right. I was confused.